war/src/main/java/io/jenkins/pluginhealth/scoring/config/SecurityConfiguration.java

```diff
@@ -34,23 +34,37 @@
 @EnableWebSecurity
 public class SecurityConfiguration {
     @Bean
-    public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
+    @Order(1)
+    public SecurityFilterChain authorizedFilterChain(HttpSecurity http) throws Exception {
         // @formatter:off
+        String [] allowedPaths = {
+            "/js/**",
+            "/style.css",
+            "/svg/**",
+            "/",
+            "/actuator/health/**",
+            "/api/scores",
+            "/data/**",
+            "/probes/**",
+            "/scores/**"
+        };
         http
-            .authorizeHttpRequests(request ->
-                request
-                    .requestMatchers(HttpMethod.GET, "/js/*", "/style.css", "/svg/*").permitAll()
-                    .requestMatchers(HttpMethod.GET,
-                        "/api/scores",
-                        "/",
-                        "/probes", "/probes/**",
-                        "/scores", "/scores/**",
-                        "/data", "/data/**",
-                        "/actuator/**"
-                    ).permitAll()
-                    .anyRequest().authenticated()
+            .securityMatcher(allowedPaths)
```

Comment:
This makes sure only path from the `allowedPaths` are considered for this configuration.
https://docs.spring.io/spring-security/reference/servlet/configuration/java.html#_choosing_securitymatcher_or_requestmatchers